homeassistant/components/automation/trace.py

```diff
@@ -204,3 +206,19 @@ def get_debug_traces(hass, summary=False):
         )
 
     return traces
+
+
+class TraceJSONEncoder(HAJSONEncoder):
+    """JSONEncoder that supports Home Assistant objects and falls back to repr(o)."""
+
+    def default(self, o: Any) -> Any:
+        """Convert certain objects.
+
+        Fall back to repr(o).
+        """
+        if isinstance(o, timedelta):
```

Comment:
Should we still do `timedelta` like this or also have our representation apply?

Reply:
We should do timedelta like that because there will be a lot of `for`, and we know they are timedeltas.
Should we do `timedelta.total_seconds()` instead though?